extensions/data-transfer/portability-data-transfer-microsoft/src/test/java/org/dataportabilityproject/transfer/microsoft/integration/MicrosoftCalendarExportTest.java

```diff
@@ -0,0 +1,256 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.dataportabilityproject.transfer.microsoft.integration;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.squareup.okhttp.HttpUrl;
+import com.squareup.okhttp.mockwebserver.MockResponse;
+import com.squareup.okhttp.mockwebserver.MockWebServer;
+import okhttp3.OkHttpClient;
+import org.dataportabilityproject.spi.transfer.provider.ExportResult;
+import org.dataportabilityproject.transfer.microsoft.calendar.MicrosoftCalendarExporter;
+import org.dataportabilityproject.transfer.microsoft.transformer.TransformerServiceImpl;
+import org.dataportabilityproject.types.transfer.auth.TokenAuthData;
+import org.dataportabilityproject.types.transfer.models.calendar.CalendarContainerResource;
+import org.junit.After;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+
+/**
+ * Verifies Calendar export using mock HTTP endpoints that replay responses from the Microsoft Graph API.
+ */
+public class MicrosoftCalendarExportTest {
+    private MockWebServer server;
+    private OkHttpClient client;
+    private ObjectMapper mapper;
+    private TransformerServiceImpl transformerService;
+    private TokenAuthData token;
+
+    @Test
+    public void testExport() throws Exception {
+        server.enqueue(new MockResponse().setBody(CALENDARS_RESPONSE));
+        server.enqueue(new MockResponse().setBody(CALENDAR1_EVENTS_RESPONSE));
+        server.enqueue(new MockResponse().setBody(CALENDAR2_EVENTS_RESPONSE));
+
+        server.start();
+
+        HttpUrl baseUrl = server.url("");
+        MicrosoftCalendarExporter exporter = new MicrosoftCalendarExporter(baseUrl.toString(), client, mapper, transformerService);
+
+        ExportResult<CalendarContainerResource> resource = exporter.export(token);
+
+        CalendarContainerResource calendarResource = resource.getExportedData();
+
+        Assert.assertEquals(2, calendarResource.getCalendars().size());
+        Assert.assertFalse(calendarResource.getCalendars().stream().anyMatch(c -> "Calendar1".equals(c.getId()) && "Calendar2".equals(c.getId())));
+
+        Assert.assertEquals(2, calendarResource.getEvents().size());
+        Assert.assertFalse(calendarResource.getEvents().stream().anyMatch(e -> "Test Appointment 1".equals(e.getTitle()) && "Test Appointment 2".equals(e.getTitle())));
+    }
+
+    @Before
+    public void setUp() {
+        client = new OkHttpClient.Builder().build();
+        mapper = new ObjectMapper();
+        transformerService = new TransformerServiceImpl();
+        token = new TokenAuthData("token123");
+        server = new MockWebServer();
+    }
+
+    @After
+    public void tearDown() throws Exception {
+        server.shutdown();
+    }
+
+    private static final String CALENDARS_RESPONSE = "{" +
```

Comment:
can these be read from a json file so we don't need all the string concatenation? fine to leave a TODO for separate PR. here & below

Reply:
I'd prefer the unit tests don't access the file system if possible but can do either way.